business/checkers/k8shttproutes/no_host_checker.go

```diff
@@ -0,0 +1,45 @@
+package k8shttproutes
+
+import (
+	"fmt"
+
+	k8s_networking_v1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/models"
+)
+
+type NoHostChecker struct {
+	Namespaces       models.Namespaces
+	K8sHTTPRoute     *k8s_networking_v1alpha2.HTTPRoute
+	RegistryServices []*kubernetes.RegistryService
+}
+
+func (n NoHostChecker) Check() ([]*models.IstioCheck, bool) {
+	validations := make([]*models.IstioCheck, 0)
+	valid := true
+	namespace := n.K8sHTTPRoute.Namespace
+
+	for k, httpRoute := range n.K8sHTTPRoute.Spec.Rules {
+		for i, ref := range httpRoute.BackendRefs {
+			if ref.Kind == nil || string(*ref.Kind) != "Service" {
+				continue
+			}
+			fqdn := kubernetes.GetHost(string(ref.Name), namespace, n.Namespaces.GetNames())
```

Comment:
For what I can see in the Object reference (https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.BackendObjectReference) It is possible to specify a namespace for the service, as an example: 

![image](https://user-images.githubusercontent.com/49480155/211804398-44f927de-9a4e-48d7-8830-b9523083b61c.png)


So in case it exist, it should be grabbed from the spec (If not, namespace should be the default): 

![image](https://user-images.githubusercontent.com/49480155/211804206-757d3c0f-9602-4306-a8f1-7ad8a5979105.png)

Reply:
Good catch, have fixed.